nmap.go

```diff
@@ -80,39 +80,53 @@ func (s *Scanner) Run() (*Run, error) {
 		done <- cmd.Wait()
 	}()
 
+	// Wait for nmap process or timeout
 	select {
 	case <-s.ctx.Done():
+
 		// Context was done before the scan was finished.
 		// The process is killed and a timeout error is returned.
 		_ = cmd.Process.Kill()
 
-		return nil, ErrScanTimeout
+		return nil, []string{}, ErrScanTimeout
```

Comment:
If the returns are named, we can directly return `warnings`, which will be an empty string slice